### PR TITLE
contrib/nodejs/globals.js: added some missing properties and methods on the global process object

### PR DESCRIPTION
**Pull request by joshtynjala**
Added a few things that were missing as documented here: https://nodejs.org/api/process.html

### CODE REVIEW

File: contrib/nodejs/globals.js [hunk 1]
```diff
@@ -104,10 +104,112 @@ function clearImmediate(id) {}
 function Process() {}
 
 /**
+ * @see http://nodejs.org/api/process.html#process_process_abort
+ */
+Process.prototype.abort = function () {};
```
Comment (MatrixFrog) [inline contrib/nodejs/globals.js]: If I remember correctly, this just tells the compiler that abort is a function, but the compiler won't know its param or return types. You need to add `@return {void}` to get the compiler to know that.

File: contrib/nodejs/globals.js [hunk 2]
```diff
@@ -104,10 +104,112 @@ function clearImmediate(id) {}
...
+/**
+ * @return {string}
+ * @see http://nodejs.org/api/process.html#process_process_cwd
+ */
+Process.prototype.cwd = function () {};
+
+/**
+ * @see http://nodejs.org/api/process.html#process_process_disconnect
+ */
+Process.prototype.disconnect = function () {};
```
Comment (MatrixFrog) [inline contrib/nodejs/globals.js]: same here

File: contrib/nodejs/globals.js (hunk 1 above)
Comment (joshtynjala) [inline contrib/nodejs/globals.js]: Added. Thanks for the tip!

File: contrib/nodejs/globals.js [hunk 3]
```diff
@@ -104,10 +104,114 @@ function clearImmediate(id) {}
...
+
+/**
+ * @param {number} pid
+ * @param {string|number} signal
+ * @see http://nodejs.org/api/process.html#process_process_kill_pid_signal
+ */
+Process.prototype.kill = function (pid, signal) {};
+
+/**
  * @param {Function} fn
```
Comment (MatrixFrog) [inline contrib/nodejs/globals.js]: Should probably be {!Function} as I suspect passing null wouldn't make much sense.

**Comment by MatrixFrog** [comment]
lgtm

**Comment by MatrixFrog** [comment]
Oh and if you can squash this into a single commit, that would be great because it makes it easier to import the change.

**Comment by Dominator008** [comment]
Copied in #2034.

**Pull request closed without merging**